contracts/svm/testing/tests/swap.rs

```diff
@@ -0,0 +1,568 @@
+        output_token.get_amount_with_decimals(8f64),
+    ));
+    assert!(Token::token_balance_matches(
+        &mut svm,
+        &output_token.get_associated_token_address(&get_express_relay_metadata_key()),
+        output_token.get_amount_with_decimals(0.08f64),
+    ));
+    assert!(Token::token_balance_matches(
+        &mut svm,
+        &output_token.get_associated_token_address(&express_relay_metadata.fee_receiver_relayer),
+        output_token.get_amount_with_decimals(0.02f64),
+    ));
+    assert!(Token::token_balance_matches(
+        &mut svm,
+        &router_output_ta,
+        output_token.get_amount_with_decimals(0.15f64),
+    ));
+}
+
+#[test]
+fn test_swap_expired_deadline() {
+    let SwapSetupResult {
+        mut svm,
+        trader,
+        searcher,
+        input_token,
+        output_token,
+        router_output_ta,
+        ..
+    } = setup_swap(SwapSetupParams {
+        platform_fee_bps:      1000,
+        input_token_program:   spl_token::ID,
+        input_token_decimals:  6,
+        output_token_program:  spl_token::ID,
+        output_token_decimals: 6,
+    });
+
+    let express_relay_metadata = get_express_relay_metadata(&mut svm);
+
+    // output token fee
+    let swap_args = SwapArgs {
+        deadline:         svm.get_sysvar::<Clock>().unix_timestamp - 1,
+        amount_input:     input_token.get_amount_with_decimals(1f64),
+        amount_output:    output_token.get_amount_with_decimals(1f64),
+        referral_fee_bps: 1500,
+        fee_token:        FeeToken::Output,
+    };
+
+    let instructions = build_swap_instructions(
+        searcher.pubkey(),
+        trader.pubkey(),
+        None,
+        None,
+        router_output_ta,
+        express_relay_metadata.fee_receiver_relayer,
+        input_token.mint,
+        output_token.mint,
+        Some(input_token.token_program),
+        Some(output_token.token_program),
+        swap_args,
+    );
+    let result =
+        submit_transaction(&mut svm, &instructions, &searcher, &[&searcher, &trader]).unwrap_err();
+    assert_custom_error(result.err, 4, ErrorCode::DeadlinePassed.into());
+
+    // now deadine is now
+    let swap_args = SwapArgs {
+        deadline:         svm.get_sysvar::<Clock>().unix_timestamp,
+        amount_input:     input_token.get_amount_with_decimals(1f64),
+        amount_output:    output_token.get_amount_with_decimals(1f64),
+        referral_fee_bps: 1500,
+        fee_token:        FeeToken::Output,
+    };
+    let instructions = build_swap_instructions(
+        searcher.pubkey(),
+        trader.pubkey(),
+        None,
+        None,
+        router_output_ta,
+        express_relay_metadata.fee_receiver_relayer,
+        input_token.mint,
+        output_token.mint,
+        Some(input_token.token_program),
+        Some(output_token.token_program),
+        swap_args,
+    );
+    submit_transaction(&mut svm, &instructions, &searcher, &[&searcher, &trader]).unwrap();
```

Comment:
you could make this `.expect`? also would prefer this in separate test

Reply:
I moved this case to the happy path